app/views/shared/_collection_menu.html.haml

```diff
@@ -13,5 +13,8 @@
         = will_paginate collection
     .action_menu_header_right
       %ul.menu
-        %li= link_to "create #{collection_class.display_name}", :controller => collection_class.name.underscore.pluralize, :action => :new
+        - if @projects && policy(@projects).create?
```

Comment:
I'd suggest getting rid of the elsif below and replace the line above with:
`!@projects || (@projects && policy(@projects).create?)`

However can you describe this better? In what case is `@projects` false?  
Perhaps this would be simplified and more secure if the check was `policy(Admin::Project).create?`

Reply:
In reviewing this, I can't see a good reason why I included that elsif and think it was most likely a mistake. I think your suggestion for simplifying things makes sense. Thanks.

Reply:
Looks like the change to `policy(@projects).create?` fails the tests.  I'd guess that `policy(Admin::Project).create?` will work better. That way even if `@projects` is nil, it can still check that the current user has permission to create a new one. But perhaps `policy(Admin::Project)` is not an allowed usage of the policy method. It is documented that the `authorize` method can be used with a class, but I didn't seen an example of the policy method being used that way.

Reply:
Thanks. `policy(Admin::Project).create? does appear to work.

I finally remembered why I had added that elsif -- `app/views/shared/_collection_menu.html.haml` is used by multiple pages, not just the Projects page. So in the case that it's being used on say the Auth Clients page, the "Create Client" link might not show up without the !@projects part of the conditional. 

Does how I included that in the latest commit look alright and make sense?

Reply:
Ah I didn't realize that the "collection" here was a generic collection of things in the portal code. After a search I can see this collection menu is used in around 20 places.  

I think a better fix would be this:
`policy(collection_class).create?`

That way whatever the policy is for the collection being shown, that policy will be used to decided what to show. I spot checked and it looks like every model that uses this collection_menu partial has a policy file. Most of them just extend ApplicationPolicy which means any logged in user can create. So that would have no effect on this new code. And the policies that do have a more restrictive create action (like the project policy in this PR) would now be respected by this partial.